### PR TITLE
Add IDBIndex cursor iteration test

### PR DESCRIPTION
**Pull request by philipwalton**
Checks for the issue reported in this Safari bug:
https://bugs.webkit.org/show_bug.cgi?id=178995

### CODE REVIEW

File: IndexedDB/idbcursor-iterating-update.htm [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    const cursorValues = [];
+    nameIndex.openCursor().onsuccess = (evt) => {
+      const cursor = evt.target.result;
+      if (cursor) {
+        func(cursor);
+        cursorValues.push(cursor.value);
+        cursor.continue();
+      } else {
+        assert_equals(cursorValues.length, 4,
+            `Expected cursor to iterate over 4 records`);
```
Comment (pwnall) [inline IndexedDB/idbcursor-iterating-update.htm]: nit: The currently preferred style is "Cursor should iterate over 4 records"

**Review by pwnall — APPROVED** [review]
LGTM w/ nit, assuming Travis passes.

Thank you very much for the test!

File: IndexedDB/idbcursor-iterating-update.htm (hunk 1 above)
Comment (philipwalton) [reply IndexedDB/idbcursor-iterating-update.htm]: Done.

**Comment by w3c-bots** [comment]
# Build PENDING

Started: None
Finished: None





View more information about this build on:

- [WPT PR Status](http://pulls.web-platform-tests.org/build/20361)
- [TravisCI](https://travis-ci.org/w3c/web-platform-tests/builds/299375491)

**Pull request merged**